sklearn/base.py

```diff
@@ -387,13 +393,25 @@ def _validate_data(self, X, y=None, reset=True,
         X : {array-like, sparse matrix, dataframe} of shape \
                 (n_samples, n_features)
             The input samples.
-        y : array-like of shape (n_samples,), default=None
-            The targets. If None, `check_array` is called on `X` and
-            `check_X_y` is called otherwise.
+        y : array-like of shape (n_samples,), default=__NO_Y
```

Comment:
lol I think this might be the only case where it would make sense to use `optional` instead of indicating `default=...`. But we can leave it as-is since it's still private.